borg/crypto.pyx

```diff
@@ -124,7 +143,7 @@ cdef class AES:
         if not out:
             raise MemoryError
         try:
-            if not EVP_DecryptUpdate(&self.ctx, out, &outl, data, inl):
+            if not EVP_DecryptUpdate(&self.ctx, out, &outl, <const unsigned char*> data_buf.buf, inl):
```

Comment:
My understanding is that Cython's `<>` operator has two fundamentally different meanings:

1. Python objects are involved, then it will do a Python-like type coercion (which will very likely make a copy)
2. The argument and the target type are C types: Same semantics as a C cast (never copies)

data_buf.buf is a `void*`, which is why a cast is needed here.